invenio_rdm_records/mappings/v6/records/record-v1.0.0.json

```diff
@@ -4,85 +4,289 @@
+              "type": "keyword",
+              "copy_to": "filename"
+            },
+            "version_id": {
+              "type": "keyword"
+            },
+            "size": {
+              "type": "long"
+            },
+            "checksum": {
               "type": "keyword"
+            },
+            "previewer": {
+              "type": "keyword"
+            },
+            "type": {
+              "type": "keyword",
+              "copy_to": "filetype"
             }
           }
         },
-        "additional_titles": {
-          "type": "nested",
+        "_default_preview": {
+          "enabled": false
+        },
+        "_internal_notes": {
+          "type": "object",
           "properties": {
-            "title": {
-              "type": "text",
-              "copy_to": "suggest_title"
+            "user": {
+              "type": "text"
+            },
+            "note": {
+              "type": "text"
+            },
+            "timestamp": {
+              "type": "date"
+            }
+          }
+        },
+        "_oai": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "keyword"
             },
-            "title_type": {
+            "sets": {
               "type": "keyword"
             },
-            "lang": {
+            "updated": {
+              "type": "date"
+            }
+          }
+        },        
+        "resource_type": {
+          "type": "object",
+          "properties": {
+            "type": {
+              "type": "keyword"
+            },
+            "subtype": {
               "type": "keyword"
             }
           }
         },
-        "contributors": {
+        "identifiers": {
           "type": "object",
           "properties": {
-            "ids": {
+            "identifier": {
+              "type": "keyword"
+            },
+            "scheme": {
+              "type": "keyword"
+            }
+          }
+        },
+        "creators": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "text"
+            },
+            "type": {
+              "type": "keyword"
+            },
+            "given_name": {
+              "type": "text"
+            },
+            "family_name": {
+              "type": "text"
+            },
+            "identifiers": {
               "type": "object",
               "properties": {
-                "source": {
-                  "type": "text",
-                  "fields": {
-                    "keyword": {
-                      "type": "keyword"
-                    }
-                  }
+                "identifier": {
+                  "type": "keyword"
                 },
-                "value": {
+                "scheme": {
                   "type": "keyword"
                 }
               }
             },
             "affiliations": {
+              "type": "object",
```

Comment:
In the PR message you mentioned not using `nested`. Could you expand on that? Unless we mark most of these as nested, we will get some strange results, no?

Reply:
I agree, however there is a slight different. On one case (`object`) we get some "not-optimal" results. As in, we might get some more results that expected. However, as of today, if we use `nested` we will **not get any result** since the current querying system does not support nested queries. So I discussed briefly with @slint and we agreed to go fully with `object` for now. And once we have support for nested we just re-index (since schema does not change it will not be too costly).

I think we should have support for nested, but it is a huge task (might involve developing a query parser) and due to time constraints we cannot do it atm (IMHO).

Reply:
What is meant by `nested` not being supported? I made queries work with it... it wasn't pretty but it worked.

Reply:
Yeah, it can be made work. But I think we should aim for something that is pretty and works. I fear if we do it no in a not pretty way, it will never be fixed...

Reply:
Ok. I think `object` will mostly work. We will see about some corner cases, but maybe nothing will come of them. :+1: